native/src/routes/Pois.tsx

```diff
@@ -87,43 +94,74 @@ const Pois = ({ pois, language, cityModel, route, navigation }: PoisProps): Reac
       }
     }, RESTORE_TIMEOUT)
   }
-  const selectPoiFeature = (feature: PoiFeature | null) => {
-    if (feature && cameraRef.current) {
-      setFollowUserLocation(false)
-      navigation.setParams({ slug: feature.properties.slug })
+
+  const deselectFeature = useCallback(() => {
+    if (multipoi && currentPoi) {
+      navigation.setParams({ slug: undefined })
+    } else {
+      navigation.setParams({ slug: undefined, multipoi: undefined })
+      scrollTo(listScrollPosition)
+    }
+  }, [currentPoi, listScrollPosition, multipoi, navigation])
+  useOnBackNavigation(slug || multipoi ? deselectFeature : undefined)
+
+  const selectFeatureOnMap = (feature: PoiFeature | null) => {
+    if (!cameraRef.current) {
+      return
+    }
+    if (!feature) {
+      deselectFeature()
+      return
+    }
+    setFollowUserLocation(false)
+    if (isMultipoi(feature)) {
+      navigation.setParams({ multipoi: feature.id as number })
       scrollTo(0)
     } else {
-      navigation.setParams({ slug: undefined })
+      const poiFeature = feature.properties.pois[0]
+      navigation.setParams({ slug: poiFeature?.slug })
+      scrollTo(0)
     }
   }
 
-  const deselectPoiFeature = useCallback(() => {
-    navigation.setParams({ slug: undefined })
-    scrollTo(listScrollPosition)
-  }, [listScrollPosition, navigation])
-  useOnBackNavigation(slug ? deselectPoiFeature : undefined)
+  const selectPoiFeatureInList = useCallback(
```

Comment:
i think you don't need a `useCallback` here if its not in a hook

Reply:
Jep you're right. I removed all unneeded callbacks.